soc/xtensa/intel_adsp/ace/multiprocessing.c

```diff
@@ -40,8 +41,18 @@ static void ipc_isr(void *arg)
 #endif
 }
 
+#define DFIDCCP			0x2020
```

Comment:
In with "register addresses in devicetree please" before someone else gets there first.

Also: is this register ACE-only?  Almost all the core system capability stuff is unchanged between cAVS and ACE (though I can't check anymore w/o docs).   I bet you anything this could be done portably.

Reply:
I plan on reworking the devicetree stuff for the config registers so will make this DT based when I do that.

As for doing this on cAVS, not sure if there is any really need/value at this point.  @nashif thoughts?

Reply:
FWIW: there are customers shipping firmware based on these drivers with many years of support promises, they'd like to keep the software as unified as possible.  :)

No seriously, there's absolutely nothing ACE-specific about this feature.  Intel's been shipping SMP audio DSPs going all the way back to Apollo Lake, and any one of them might very well want to auto-detect CPU counts (IIRC this same situation already exists in exactly this form on TGL, where there are two variant implementations with either 2 or 4 cores, no?)

Reply:
>  where there are two variant implementations with either 2 or 4 cores, no?)

Yes:

```shell
$ cd sof
$ ./scripts/xtensa-build-all.sh tgl tgl-h

$ grep CORE_COUNT build_tgl*/generated/.config

build_tgl_gcc/generated/.config:CONFIG_MAX_CORE_COUNT=4
build_tgl_gcc/generated/.config:CONFIG_CORE_COUNT=4
build_tgl-h_gcc/generated/.config:CONFIG_MAX_CORE_COUNT=4
build_tgl-h_gcc/generated/.config:CONFIG_CORE_COUNT=2

```

Reply:
TGL having different core counts is already covered with multiple platform definitions like above, starting with ACE, we do not want to do that. 
It would be interesting how this would work on CAVS actually, especially on TGL, this could be a our upstream test platform for this feature, so it is worth looking into that, but we do not need to make this backward compatible if the case is covered in some other way.